### PR TITLE
[#2634] Allow delete dataset columns tx if there is no visualisations related

### PR DESCRIPTION
**Pull request by tangrammer**
- [ ] **Update release notes if necessary**
Relates #2634 

- could be improved with specs the same way we do with authorization => created issue https://github.com/akvo/akvo-lumen/issues/2645

UI message after changes
Screenshot 2020-04-08 at 11 04 06

### CODE REVIEW

**Comment by tangrammer** [comment]
@iperdomo the review could be done following commit changes 🤓

File: backend/src/akvo/lumen/lib/transformation/delete_column.clj [hunk 1]
```diff
@@ -21,17 +24,33 @@
...
     (if (empty? merged-sources)
-      (let [column-idx  (engine/column-index columns column-name)]
-        (db.tx.engine/delete-column tenant-conn {:table-name table-name :column-name column-name})
-        {:success?      true
-         :execution-log [(format "Deleted column %s" column-name)]
-         :columns       (into (vec (take column-idx columns))
-                              (drop (inc column-idx) columns))})
+      (if-let [existent-viss (seq (visualisations-with-dataset-column tenant-conn (:dataset-id op-spec) column-name))]        
+        {:success? false
+         :message  (format "Cannot delete column. It is used in the following visalisations: %s"
```
Comment (iperdomo) [inline backend/src/akvo/lumen/lib/transformation/delete_column.clj:44]: This user message needs to be _localized_. How are we going to do that?

Comment (tangrammer) [reply backend/src/akvo/lumen/lib/transformation/delete_column.clj:44]: https://github.com/akvo/akvo-lumen/issues/2646

**Comment by tangrammer** [comment]
@iperdomo merging PR after creating a different issue for the translation of backend user messages. Also changed the PR title to better specify the PR changes

**Pull request merged**